bundle/compliance/cis_eks/rules/cis_3_2_10/test.rego

```diff
@@ -4,34 +4,38 @@ import data.compliance.cis_eks.data_adapter
 import data.kubernetes_common.test_data
 import data.lib.test
 
-violations {
-	test.assert_fail(finding) with input as rule_input("--rotate-certificates false")
-	test.assert_fail(finding) with input as rule_input_with_external("", create_process_config(false))
-	test.assert_fail(finding) with input as rule_input_with_external("--rotate-certificates false", create_process_config(true))
-}
-
-test_violations {
-	violations with data.benchmark_data_adapter as data_adapter
-}
-
-passes {
-	test.assert_pass(finding) with input as rule_input("")
-	test.assert_pass(finding) with input as rule_input("--rotate-certificates true")
-	test.assert_pass(finding) with input as rule_input_with_external("--rotate-certificates true", create_process_config(false))
-	test.assert_pass(finding) with input as rule_input_with_external("--rotate-certificates true", create_process_config(true))
-	test.assert_pass(finding) with input as rule_input_with_external("", create_process_config(true))
+test_violation {
+	eval_fail with input as rule_input("--rotate-certificates false")
+	eval_fail with input as rule_input_with_external("", create_process_config(false))
+	eval_fail with input as rule_input_with_external("--rotate-certificates false", create_process_config(true))
 }
 
 test_pass {
-	passes with data.benchmark_data_adapter as data_adapter
+	eval_pass with input as rule_input("")
+	eval_pass with input as rule_input("--rotate-certificates true")
+	eval_pass with input as rule_input_with_external("--rotate-certificates true", create_process_config(false))
+	eval_pass with input as rule_input_with_external("--rotate-certificates true", create_process_config(true))
+	eval_pass with input as rule_input_with_external("", create_process_config(true))
 }
 
 test_not_evaluated {
-	not finding with input as test_data.process_input("some_process", []) with data.benchmark_data_adapter as data_adapter
+	not_eval with input as test_data.process_input("some_process", [])
 }
 
 rule_input(argument) = test_data.process_input("kubelet", [argument])
 
 rule_input_with_external(argument, external_data) = test_data.process_input_with_external_data("kubelet", [argument], external_data)
 
 create_process_config(rotateCertificates) = {"config": {"rotateCertificates": rotateCertificates}}
+
+eval_fail {
+	test.assert_fail(finding) with data.benchmark_data_adapter as data_adapter
+}
+
+eval_pass {
+	test.assert_pass(finding) with data.benchmark_data_adapter as data_adapter
+}
+
+not_eval {
+	not finding with data.benchmark_data_adapter as data_adapter
+}
```

Comment:
can this be common?

Reply:
like a function?

Reply:
not sure if it will affect performance though

Reply:
Moving it and creating a common flow is a bit problematic.
Because there is no `input` (it is being assigned in every test case) and the `benchmark data adapter` expects the input to be available upon import in order to work properly.
The `finding` also expects both of these flows to happen prior to being called.
This creates a dependency situation that requires the following flow:
`input` -> `benchmark data adapter` -> `finding`.

I've attempted to use the `with` keyword and create this dependency by piping the `with` commands, doesn't seem like it works.

I've also attempted to create a common function that gets the finding and the benchmark adapter and is being called with `input` using `with` from outside.
Sadly, `Rego` doesn't allow assigning variables into `with`:
`rego_type_error: with keyword target must reference existing input, data, or a function`.
But even if that would work, there are still problems regarding passing `finding` to the function because in order to be passed it is evaluated immediately, which wouldn't work because there is no input and data benchmark in its scope.

There is also an option of wrapping all of the benchmark data adapters in `functions` instead of using them by `import` relations.
That will require changes to the code base, and not sure if that is a better coding solution than what we have right now with the imports (both performance-wise and in understanding the code flows).

I agree that this is something that needs to be investigated and I think that we should take a deeper look at how to resolve this, instead of us copying that over and over for every test file.